File: test/fixtures/ruby/post_json_multiple_headers.rb
require 'net/http'
require 'json'
require 'openssl'

uri = URI('http://localhost:28139/rest/login-sessions')
req = Net::HTTP::Post.new(uri)
req.content_type = 'application/json'
req['X-API-Version'] = '200'

# The object won't be serialized exactly like this
# req.body = '{"userName":"username123","password":"password123", "authLoginDomain":"local"}'
req.body = {
  'userName' => 'username123',
  'password' => 'password123',
  'authLoginDomain' => 'local'
}.to_json

req_options = {
  use_ssl: uri.scheme == 'https',
  verify_mode: OpenSSL::SSL::VERIFY_NONE
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/r-httr2/post_json_multiple_headers.r
library(httr2)

request("http://localhost:28139/rest/login-sessions") |>
  req_method("POST") |>
  req_headers(`X-API-Version` = "200") |>
  req_body_raw(
    '{"userName":"username123","password":"password123", "authLoginDomain":"local"}',
    type = "application/json"
  ) |>
  req_options(ssl_verifypeer = 0) |>
  req_perform(verbosity = 1)
